sdk/core/core-amqp/src/errors.ts

```diff
@@ -548,46 +568,43 @@ function isBrowserWebsocketError(err: any): boolean {
 }
 
 /**
- * Translates the AQMP error received at the protocol layer or a generic Error into a MessagingError.
+ * Translates the AQMP error received at the protocol layer or a SystemError into a MessagingError.
+ * All other errors are returned unaltered.
  *
  * @param {AmqpError} err The amqp error that was received.
  * @returns {MessagingError} MessagingError object.
  */
-export function translate(err: AmqpError | Error): MessagingError {
+export function translate(err: AmqpError | Error): MessagingError | Error {
   if ((err as MessagingError).translated) {
     // already translated
     return err as MessagingError;
   }
 
-  let error: MessagingError = err as MessagingError;
-
   // Built-in errors like TypeError and RangeError should not be retryable as these indicate issues
   // with user input and not an issue with the Messaging process.
   if (err instanceof TypeError || err instanceof RangeError) {
-    error.retryable = false;
-    return error;
+    return err;
   }
 
   if (isAmqpError(err)) {
     // translate
     const condition = (err as AmqpError).condition;
     const description = (err as AmqpError).description as string;
-    error = new MessagingError(description);
+    const error = new MessagingError(description);
     if ((err as any).stack) error.stack = (err as any).stack;
     error.info = (err as AmqpError).info;
-    error.condition = condition;
+    error.amqpCondition = condition;
     if (condition) {
-      error.name = ConditionErrorNameMapper[condition as keyof typeof ConditionErrorNameMapper];
+      error.code = ConditionErrorNameMapper[condition as keyof typeof ConditionErrorNameMapper];
     }
-    if (!error.name) error.name = "MessagingError";
     if (
       description &&
       (description.includes("status-code: 404") ||
         description.match(/The messaging entity .* could not be found.*/i) !== null)
     ) {
-      error.name = "MessagingEntityNotFoundError";
+      error.code = "MessagingEntityNotFoundError";
     }
-    if (retryableErrors.indexOf(error.name) === -1) {
+    if (error.code && retryableErrors.indexOf(error.code) === -1) {
```

Comment:
What if by this point we don't have a `code`?

Reply:
At this point I was treating errors without a code as neither an `AmqpError` nor a `SystemError`, so do not alter them. They currently don't have a `retryable` field so are effectively not retryable.

This could change based on the thread you started at https://github.com/Azure/azure-sdk-for-js/pull/6673/files#r362364248

Reply:
Prior to this PR, no amqp condition results in the translated error having the name `MessagingError` and the `retryable` property set to `false`.

Post this PR, no amqp condition results in the translated error having no code (which is fine), but with `retryable` property set to `true`.

While `isAmqpError()` ensures that the error has the condition property, this code path will currently be hit if there was an error with an amqp condition for which we have no mapping.

Reply:
Currently, if there isn't a code, we will treat the error as retryable since retryable is set to true in the MessagingError constructor and we don't change it to false here. That seems consistent with the behavior prior to this PR, since if no condition was found we'd set the `name` to `MessagingError`, and any error with the name `MessagingError` was treated as retryable.

So, I think the question you're getting at is whether we should set `error.code` if there is no mapping or not. My preference would be to not do this, because then it would be a breaking change if we wanted to change the code in the future for an error we learned more about but weren't handling before. I __think__ we can safely add a code if it never existed in the 1st place, but not change it if there already was one for an error.

Reply:
> That seems consistent with the behavior prior to this PR, since if no condition was found we'd set the name to MessagingError, and any error with the name MessagingError was treated as retryable.

Not true... `retryableErrors` would not have an entry for `MessagingError` and so `retryable` would be set too false...

Reply:
https://github.com/Azure/azure-sdk-for-js/blob/4b0f3ad5a8b5371e88af1a25f15a3baf5d993995/sdk/core/core-amqp/src/errors.ts#L490

retryableErrors does have an entry for MessagingError today though...

Reply:
ah! I missed that :)